internal/security/proxy/service.go

```diff
@@ -0,0 +1,332 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ * @version: 1.1.0
+ *******************************************************************************/
+package proxy
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/dghubble/sling"
+	"net/http"
+)
+
+type Service struct {
+	Connect    Requestor
+	CertCfg    CertConfig
+	ServiceCfg ServiceConfig
+}
+
+type ServiceConfig interface {
```

Comment:
This interface represents configuration for service. In the unit testing these functions can be implemented in a way that a remote server is not needed.

Reply:
Same comment as above... it doesn't seem to be an appropriate usage of interfaces in Go since all the methods just return a specific thing from the struct that could just as easily be mocked by setting a different property in the struct